File: Script Counter CS V3/Options.cs
#nullable disable
using Newtonsoft.Json;

namespace ScriptCounterV3
{
    public class Options
    {
        public const string FilePath = "Options.json";

        public List<string> FileTypes = new();
        public List<string> PathFilters = new();

        public static Options Get()
        {
            if (File.Exists(FilePath))
            {
                return JsonConvert.DeserializeObject<Options>(File.ReadAllText(FilePath));
            }
            var options = new Options();
            options.FileTypes.Add("*.cs");
            File.WriteAllText(FilePath, JsonConvert.SerializeObject(options));
            return options;
        }
    }
}

File: Script Counter Cpp V2/Options.cpp
#include "Options.h"

bool Options::IsEmpty()
{
	if (FileTypes.empty()) 
	{
		return true;
	}
	return false;
}

bool Options::Load() 
{
	if (std::filesystem::exists(FilePath))
	{
		std::ifstream file;
		file.open(FilePath);
		bool state = file.is_open();
		if (state)
		{
			nlohmann::json j = nlohmann::json::parse(file);
			FileTypes = j["file types"];
			PathFilters = j["path filters"];
		}
		file.close();
		return state;
	}
	return false;
}

void Options::Save()
{
	nlohmann::json j;
	j["file types"] = FileTypes;
	j["path filters"] = PathFilters;

	std::ofstream file(FilePath);
	file << j.dump(4) << std::endl;
	file.close();
}